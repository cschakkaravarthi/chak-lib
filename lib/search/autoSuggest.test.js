"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
var react_1 = __importDefault(require("react"));
var testHelper_1 = require("../services/testHelper");
var autoSuggest_1 = require("./autoSuggest");
describe('SearchBarAutoSuggest', function () {
    it('should render correctly', function () {
        var tree = testHelper_1.shallowRender(react_1["default"].createElement(autoSuggest_1.AutoSuggest, { handleSubmit: function () { alert('clicked'); }, suggestions: [], getSuggestions: function () { alert('clicked'); }, handleSelected: function () { alert('clicked'); }, manualSearch: function () { alert('clicked'); }, miniNoOfCharsShowSuggestions: 2, onSuggestionsClearRequested: function () { alert('clicked'); }, suggestionValue: "" }));
        expect(tree).toMatchSnapshot();
    });
    it('should render correctly mobile search bar', function () {
        var tree = testHelper_1.shallowRender(react_1["default"].createElement(autoSuggest_1.AutoSuggest, { handleSubmit: function () { alert('clicked'); }, suggestions: [], getSuggestions: function () { alert('clicked'); }, handleSelected: function () { alert('clicked'); }, manualSearch: function () { alert('clicked'); }, miniNoOfCharsShowSuggestions: 2, onSuggestionsClearRequested: function () { alert('clicked'); }, suggestionValue: "", isMobileSearchBar: true }));
        expect(tree).toMatchSnapshot();
    });
});
//# sourceMappingURL=autoSuggest.test.js.map