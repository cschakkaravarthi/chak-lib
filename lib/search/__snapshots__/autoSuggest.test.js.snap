// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBarAutoSuggest should render correctly 1`] = `
<ForwardRef
  className="auto-suggest-form align-items-center d-flex"
  inline={false}
  onReset={[Function]}
  onSubmit={[Function]}
>
  <ForwardRef>
    <ForwardRef>
      <ForwardRef
        active={false}
        className="pl-0 border-0"
        disabled={false}
        type="submit"
        variant="outline-secondary"
      >
        <img
          className="search-icon"
          src="test-file-stub"
        />
      </ForwardRef>
    </ForwardRef>
    <Autosuggest
      alwaysRenderSuggestions={false}
      focusInputOnSuggestionClick={true}
      getSuggestionValue={[Function]}
      highlightFirstSuggestion={false}
      id="1"
      inputProps={
        Object {
          "onChange": [Function],
          "placeholder": undefined,
          "value": "",
        }
      }
      multiSection={false}
      onSuggestionSelected={[Function]}
      onSuggestionsClearRequested={[Function]}
      onSuggestionsFetchRequested={[Function]}
      renderSuggestion={[Function]}
      renderSuggestionsContainer={[Function]}
      shouldRenderSuggestions={[Function]}
      suggestions={Array []}
      theme={
        Object {
          "container": "react-autosuggest__container",
          "containerOpen": "react-autosuggest__container--open",
          "input": "react-autosuggest__input",
          "inputFocused": "react-autosuggest__input--focused",
          "inputOpen": "react-autosuggest__input--open",
          "sectionContainer": "react-autosuggest__section-container",
          "sectionContainerFirst": "react-autosuggest__section-container--first",
          "sectionTitle": "react-autosuggest__section-title",
          "suggestion": "react-autosuggest__suggestion",
          "suggestionFirst": "react-autosuggest__suggestion--first",
          "suggestionHighlighted": "react-autosuggest__suggestion--highlighted",
          "suggestionsContainer": "react-autosuggest__suggestions-container",
          "suggestionsContainerOpen": "react-autosuggest__suggestions-container--open",
          "suggestionsList": "react-autosuggest__suggestions-list",
        }
      }
    />
    <div
      className="d-none d-lg-inline"
    />
  </ForwardRef>
</ForwardRef>
`;

exports[`SearchBarAutoSuggest should render correctly mobile search bar 1`] = `
<ForwardRef
  className="mobile-search-bar align-items-center d-flex"
  inline={false}
  onReset={[Function]}
  onSubmit={[Function]}
>
  <ForwardRef>
    <Autosuggest
      alwaysRenderSuggestions={false}
      focusInputOnSuggestionClick={true}
      getSuggestionValue={[Function]}
      highlightFirstSuggestion={false}
      id="1"
      inputProps={
        Object {
          "onChange": [Function],
          "placeholder": undefined,
          "value": "",
        }
      }
      multiSection={false}
      onSuggestionSelected={[Function]}
      onSuggestionsClearRequested={[Function]}
      onSuggestionsFetchRequested={[Function]}
      renderSuggestion={[Function]}
      renderSuggestionsContainer={[Function]}
      shouldRenderSuggestions={[Function]}
      suggestions={Array []}
      theme={
        Object {
          "container": "react-autosuggest__container",
          "containerOpen": "react-autosuggest__container--open",
          "input": "react-autosuggest__input",
          "inputFocused": "react-autosuggest__input--focused",
          "inputOpen": "react-autosuggest__input--open",
          "sectionContainer": "react-autosuggest__section-container",
          "sectionContainerFirst": "react-autosuggest__section-container--first",
          "sectionTitle": "react-autosuggest__section-title",
          "suggestion": "react-autosuggest__suggestion",
          "suggestionFirst": "react-autosuggest__suggestion--first",
          "suggestionHighlighted": "react-autosuggest__suggestion--highlighted",
          "suggestionsContainer": "react-autosuggest__suggestions-container",
          "suggestionsContainerOpen": "react-autosuggest__suggestions-container--open",
          "suggestionsList": "react-autosuggest__suggestions-list",
        }
      }
    />
    <div
      className=" d-lg-inline"
    />
    <ForwardRef>
      <ForwardRef
        active={false}
        className="p-1 border-0"
        disabled={false}
        type="submit"
        variant="outline-light"
      >
        <img
          className="search-icon"
          src="test-file-stub"
        />
      </ForwardRef>
    </ForwardRef>
  </ForwardRef>
</ForwardRef>
`;
